api/user.js

```diff
@@ -17,4 +17,86 @@ usersRouter.get('/', async (req,res,next) => {
     }
 })
 
+usersRouter.post("/login", async (req, res, next) => {
+  const { username, password } = req.body;
+
+  // request must have both
+  if (!username || !password) {
+    next({
+      name: "MissingCredentialsError",
+      message: "Please supply both a username and password",
+    });
+  }
+
+  try {
+    const user = await getUserByUsername(username);
```

Comment:
These functions won't work as written because of how they're being imported from the db. It would need to be written as User.getUserByUsername